tests/integrations/test_feature_dynamics_extraction_with_selection.py

```diff
@@ -0,0 +1,1268 @@
+            column_id=column_params_config["column_id"],
+            column_sort=column_params_config["column_sort"],
+            column_kind=column_params_config["column_kind"],
+            column_value=column_params_config["column_value"],
+        )
+
+        some_expected_feature_dynamics_names = (
+            'D_dt_y1dt_y2||quantile||q_0.2@window_5__fft_coefficient__attr_"real"__coeff_1',
+            "D_dt_y1y3||number_cwt_peaks||n_3@window_5__number_cwt_peaks__n_3",
+            "y2||permutation_entropy||dimension_2||tau_1@window_4__quantile__q_0.2",
+            'D_y2y3||fft_coefficient||attr_"real"||coeff_1@window_4__number_cwt_peaks__n_3',
+        )
+
+        self.assertIsInstance(X, pd.DataFrame)
+        self.assertTrue(len(X) == 10)
+        self.assertTrue(
+            set(some_expected_feature_dynamics_names).issubset(X.columns.tolist())
+        )
+        # We cant make strong claims about the number of different features produced because some feature timeseries have NaNs and are dropped
+
+        # c) Select
+        X_relevant = select_features(X, response, fdr_level=0.999)
+
+        # d) Gen relevant features dictionaries
+        rel_feature_names = list(X_relevant.columns)
+        (
+            rel_feature_time_series_dict,
+            rel_feature_dynamics_dict,
+        ) = derive_features_dictionaries(rel_feature_names)
+
+        # e) Relevant features interpretation
+        output_filename_prefix = "feature_dynamics_interpretation_test"
+        gen_pdf_for_feature_dynamics(
+            rel_feature_names, output_filename=output_filename_prefix
+        )
+
+        pdf_exists = os.path.exists(f"{output_filename_prefix}.pdf")
+        markdown_exists = os.path.exists(f"{output_filename_prefix}.md")
+
+        if pdf_exists:
+            os.remove(f"{output_filename_prefix}.pdf")
+        if markdown_exists:
+            os.remove(f"{output_filename_prefix}.md")
+
+        # f) extract on selected features
+        X_more = extract_feature_dynamics(
+            timeseries_container=ts_with_extra_timeseries_between_and_within,
+            n_jobs=0,
+            feature_timeseries_kind_to_fc_parameters=rel_feature_time_series_dict,
+            feature_dynamics_kind_to_fc_parameters=rel_feature_dynamics_dict,
+            column_id=column_params_config["column_id"],
+            column_sort=column_params_config["column_sort"],
+            column_kind=column_params_config["column_kind"],
+            column_value=column_params_config["column_value"],
+        )
+
+        self.assertIsInstance(X_more, pd.DataFrame)
+        self.assertTrue(len(X_more) == 10)
+        # Check that feature vectors are the same no matter how they are extracted
+        for feature_name in X_more.columns:
+            pd.testing.assert_series_equal(
+                X_more[feature_name], X[feature_name]
+            )  # checking idempotency
+
+    def test_full_feature_dynamics_workflow_pandas_dict(self):
```

Comment:
Line 1051: @kempa-liehr this is the function that has the following bug: The same dataset is used in a different format, and different features are extracted compared to when a different format is used